src/domains/lattice.ml

```diff
@@ -581,6 +581,33 @@ struct
     Pretty.dprintf "%a not leq %a" pretty x pretty y
 end
 
+module type Num = sig val x : unit -> int end
+module ProdList (Base: S) (N: Num) =
+struct
+  include Printable.Liszt (Base)
+
+  let bot () = Array.to_list (Array.make (N.x ()) (Base.bot ()))
+  let is_bot =
+    let f acc x = Base.is_bot x && acc in
+    List.fold_left f true
```

Comment:
This is just `List.for_all`.

Reply:
Fixed on cddbe1a915f6704d7226f549abac76bdadf73bba